### PR TITLE
Loader: Move MissingTest mapping to TestLoaderProxy [v4]

### PR DESCRIPTION
**Pull request by ldoktor**
The MissingTest is yielded by TestLoaderProxy, but it was implemented
and described in FileLoader, which caused troubles when FileLoader was
not being used. This simple fix allows TestLoaderProxy to define basic
mappings and adds MissingTest to it.

v1: https://github.com/avocado-framework/avocado/pull/1920
v2: https://github.com/avocado-framework/avocado/pull/1926
v3: https://github.com/avocado-framework/avocado/pull/1933

Changes:
```yaml
v2: Use different approach as this is a bugfix and the correct fix will be introduced later
v3: Rework the lazy-init to normal init on `load_plugins`
v3: Add selftest
v4: Add new commit to fix uncovered branch making FileLoader to crash
v4: Move the update_mappings to a different function to improve readability
v4: Remove unnecessary accidental change in `avocado.plugins.list`
```

### CODE REVIEW

**Comment by ldoktor** [comment]
[BUG: The `avocado list` is broken when `file` loader is not enabled](https://trello.com/c/r4DLYomE/982-bug-the-avocado-list-is-broken-when-file-loader-is-not-enabled)

File: avocado/core/loader.py [hunk 1]
```diff
@@ -207,16 +223,10 @@ def get_base_keywords(self):
         return base_path
 
     def get_type_label_mapping(self):
-        mapping = {}
-        for loader_plugin in self._initialized_plugins:
-            mapping.update(loader_plugin.get_type_label_mapping())
-        return mapping
```
Comment (apahim) [inline avocado/core/loader.py]: Why did you remove this code from here?

Comment (ldoktor) [inline avocado/core/loader.py]: Because it's updated on `load_plugins`

Comment (apahim) [inline avocado/core/loader.py]: That's quite obvious, but does not really answer the question.
This code was here in your v2 (#1926) and was moved in v3 and v4. Since I already pointed this out in the v3 and you kept it in v4 (actually it's worst in v4, with a new private function), I'd like to understand the reason for that.

Comment (ldoktor) [inline avocado/core/loader.py]: The mapping depends only on `_initialized_plugins` and is used several times. It makes no sense to evaluate it every time we call the `get_*` function so I believe it's better to do this when `_initialized_plugins` is updated.

Basically it's a question whether we want to initialize this lazily, or right away. I believe the right away is better in this case as we know for sure they are static.

Comment (apahim) [inline avocado/core/loader.py]: Ok then. Thank you.
